### PR TITLE
add more helpful error for calling a decl that exists in a module

### LINKED ISSUE #6750: Unhelpful error message when you `use` a module but forget the `*`
### Describe the bug

When you try to use a module's exported command but either A) forgot to include the module namespace, or B) forgot to `use` the module without namespacing, you get an unhelpful and contradictory error.

### How to reproduce

Here's an example of what I mean:
```
C:\Users\Leon〉module greetings {
:::      export def hello [name: string] {
:::          $"hello ($name)!"
:::      }
:::
:::      export def hi [where: string] {
:::          $"hi ($where)!"
:::      }
::: }
C:\Users\Leon〉use greetings
C:\Users\Leon〉hi
Error: nu::shell::external_command (link)

  × External command failed
   ╭─[entry #8:1:1]
 1 │ hi
   · ─┬
   ·  ╰── did you mean 'hi'?
   ╰────
  help: program not found
```

### Expected behavior

Ideally, this error would tell you to write `greetings hi` instead.

### Screenshots

_No response_

### Configuration

| key                | value                                           |
| ------------------ | ----------------------------------------------- |
| version            | 0.69.1                                          |
| branch             |                                                 |
| commit_hash        | 8a5abc7afc28a36908a32f1852127e2a5170b461        |
| build_os           | windows-x86_64                                  |
| build_target       | x86_64-pc-windows-msvc                          |
| rust_version       | rustc 1.64.0 (a55dd71d5 2022-09-19)             |
| rust_channel       | stable-x86_64-pc-windows-msvc                   |
| cargo_version      | cargo 1.64.0 (387270bc7 2022-09-16)             |
| pkg_version        | 0.69.1                                          |
| build_time         | 2022-09-28 02:58:08 +00:00                      |
| build_rust_channel | release                                         |
| features           | database, dataframe, default, trash, which, zip |
| installed_plugins  |                                                 |

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by merelymyself**
# Description

Fixes #6750.
New error:
```
~/CodingProjects/nushell〉module spam { export def foo [] { "foo" } }; use spam; foo
Error: nu::shell::external_command (link)

  × External command failed
   ╭─[entry #1:1:1]
 1 │ module spam { export def foo [] { "foo" } }; use spam; foo
   ·                                                        ─┬─
   ·                                                         ╰── command 'foo' was not found but it exists in module 'spam'
   ╰────
  help: No such file or directory (os error 2)
```

# Tests

Make sure you've done the following:

- [x] Add tests that cover your changes, either in the command examples, the crate/tests folder, or in the /tests folder.
- [ ] Try to think about corner cases and various ways how your changes could break. Cover them with tests.
- [ ] If adding tests is not possible, please document in the PR body a minimal example with steps on how to reproduce so one can verify your change works.

Make sure you've run and fixed any issues with these commands:

- [x] `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- [x] `cargo clippy --workspace --features=extra -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
- [x] `cargo test --workspace --features=extra` to check that all the tests pass

### CODE REVIEW

File: crates/nu-command/src/system/run_external.rs [hunk 1]
```diff
@@ -269,6 +269,14 @@ impl ExternalCommand {
                                         "'{}' was not found, did you mean '{s}'?",
                                         self.name.item
                                     )
+                                } else if self.name.item == s {
+                                    let sugg = engine_state.which_module_has_decl(s.as_bytes());
+                                    if let Some(sugg) = sugg {
+                                        let sugg = String::from_utf8_lossy(sugg);
+                                        format!("did you mean '{sugg} {s}'?")
+                                    } else {
+                                        format!("did you mean '<module> {s}'?")
```
Comment (kubouch) [inline crates/nu-command/src/system/run_external.rs:278]: I'm thinking this might not be desired: If the command is not in a module, the suggestion should be just the default "did you mean", IMO.

File: crates/nu-command/src/system/run_external.rs [hunk 2]
```diff
@@ -269,6 +269,14 @@ impl ExternalCommand {
                                         "'{}' was not found, did you mean '{s}'?",
                                         self.name.item
                                     )
+                                } else if self.name.item == s {
+                                    let sugg = engine_state.which_module_has_decl(s.as_bytes());
+                                    if let Some(sugg) = sugg {
+                                        let sugg = String::from_utf8_lossy(sugg);
+                                        format!("did you mean '{sugg} {s}'?")
```
Comment (kubouch) [inline crates/nu-command/src/system/run_external.rs:276]: I'd rephrase it as "Command 'foo' was not found but it exists in a module 'bar'" or something like that. The "did you mean 'bar foo'?" suggestion might suggest there exists a command "bar foo" which is not true: You'd need to import it first.

**Comment by merelymyself** [comment]
Thanks for the suggestions @kubouch . Do you mind taking another look? Thanks.

File: crates/nu-command/src/system/run_external.rs [hunk 3]
```diff
@@ -269,6 +269,14 @@ impl ExternalCommand {
                                         "'{}' was not found, did you mean '{s}'?",
                                         self.name.item
                                     )
+                                } else if self.name.item == s {
```
Comment (kubouch) [inline crates/nu-command/src/system/run_external.rs:272]: Just one more question: What exactly does this condition test?

**Comment by jntrnr** [comment]
I like the idea of improving the error message. I think the one change I might think to do, if we can, is to use a message like:

```
command 'foo' was not found but it exists in module 'spam' using `spam foo`
```

Just to give people a little more hint as to what they should type to fix the error.

**Comment by merelymyself** [comment]
Done, thanks for the suggestion @jntrnr .

**Comment by fdncred** [comment]
thanks for working with us @merelymyself. let's move forward with this.
